### PR TITLE
Uki support

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

File: internal/utils/common.go [hunk 1]
```diff
@@ -108,16 +108,20 @@ func GetTarget(dryRun bool) (string, string) {
...
 
-	img = ReadCMDLineArg("cos-img/filename=")[0]
+	imgs := ReadCMDLineArg("cos-img/filename=")
 
 	// If no image just panic here, we cannot longer continue
-	if img == "" {
-		log.Logger.Fatal().Msg("Could not get the image name from cmdline (i.e. cos-img/filename=/cOS/active.img)")
+	if len(imgs) == 0 {
+		if IsUKI() {
+			imgs = []string{""}
```
Comment (mudler) [inline internal/utils/common.go:116]: small nit: this seems to be just forced to return an empty string later on?

File: pkg/mount/dag_uki_boot.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// RegisterUKI registers the dag for booting from UKI
+func (s *State) RegisterUKI(g *herd.Graph) error {
+	// Write sentinel
+	s.LogIfError(s.WriteSentinelDagStep(g), "sentinel")
+
+	s.LogIfError(s.LoadKernelModules(g), "kernel modules")
+
+	// Udev for devices discovery
+	s.LogIfError(s.UKIUdevDaemon(g), "udev")
```
Comment (mudler) [inline pkg/mount/dag_uki_boot.go:16]: I wonder, here we need just for udev rules? or is just for having disk by label available ?

**Review by mudler — APPROVED** [review]
Looking good! Awesome job! just small questions around the code, but nothing preventing merging

File: pkg/mount/dag_uki_boot.go (hunk 2 above)
Comment (Itxaka) [reply pkg/mount/dag_uki_boot.go:16]: Indeed, just for easy identification of disk by label. Could be moved into using whatever we want, but for ease of use and to reuse the steps from other paths, having the disks set up by label its the easieast way of finding them in the system

File: internal/utils/common.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ func BootStateToLabelDevice() string {
 func GetRootDir() string {
 	cmdline, _ := os.ReadFile("/proc/cmdline")
 	switch {
-	case strings.Contains(string(cmdline), "IMMUCORE_NOPIVOT"):
+	case strings.Contains(string(cmdline), "rd.immucore.uki"):
```
[the remark this answers is not in the document]
Comment (Itxaka) [reply internal/utils/common.go:35]: :+1:

File: internal/utils/common.go (hunk 1 above)
Comment (Itxaka) [reply internal/utils/common.go:116]: yeah this is done as a workaround to use the same state all across the different paths. I could work around this tbh but was the easiet way of fixing this for now.

**Comment by Itxaka** [comment]
Im stupid, didnt wat this merged without cleaning it before lol

**Pull request merged**